### ISSUE TITLE
.NET Core sln template has issues referencing some assemblies

### ISSUE THREAD

**Issue by rockfordlhotka** [aspnet/AspNetCore#5498]
The .NET Core hosted template has an issue where it can't reference some assemblies or NuGet packages.

Suppose you have a component that has the same assembly name (like `x.dll`) but is built once for wasm and once for .NET Core - so there are two assemblies named `x.dll` _but they are different builds of the same code_.

(This scenario might seem odd - but it isn't uncommon, especially with something like Blazor where the mono runtime on the client is not the same as the netcore runtime on the server. The solution is to have a codebase that is "the same" except with compiler directives to fix the differences between platforms. Yes, netstandard _should_ fix this, but is not a perfect solution.)

Right now, if you reference the mono-built `x.dll` in the Blazor app, and the netcore-built `x.dll` in the server app, the server app will fail to build.

**Comment by danroth27** [comment]
Hi @rockfordlhotka 

We don't have a TFM for mono.wasm. We currently just use .NET Standard to target mono.wasm. So you could build your component library to target both .NET Standard and .NET Core. Is that what you're currently doing?

As for the build failure your seeing, it sounds like that happens because your are using two different builds of the same assembly in the same app. Why would you need to do that?

**Comment by rockfordlhotka** [comment]
At the time I created this issue it was impossible to get all the same behaviors without using compiler directives - hence a different assembly for Blazor from the .NET Standard one on the server.

Subsequently I figured out a way to refactor the code to live within the Blazor limitations, and implemented that code for .NET Standard as well.

As long as the mono-wasm implementation of .NET Standard isn't too leaky going forward we should be fine.
